Footer Component Added

naveen8801 opened the pull request:
### Description

New Footer component is added in Home folder:-

- Added Footer.jsx and Footer.css in Home
- Added Fonts (2 fonts) and icons ( 4 icons ) in assets folder

![latest_footer](https://user-images.githubusercontent.com/54985099/115994165-75632580-a5f3-11eb-9159-de363ff9d0eb.gif)



Fixes #215

### Type of Change:

- User Interface

**Code/Quality Assurance Only**

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:
<!--**Delete irrelevant options.**-->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

naveen8801 committed:
Footer Component Added

shades-7 commented:
@mtreacy002  This is the first time I am trying to test a PR. so let me know if i am doing it wrong.
![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/58786336/114439571-be27e100-9be6-11eb-8ece-4d1ae5ad29c7.gif)

This is my feedbacks:
* All the social icons are directing to the same page.
* Line spacing is more for this paragraph  "We are known for connecting top talents with top comapnies. We make great possibilities happen." 
* In mobile view stacking is done but i think stacking need to de done in all the subsections too like explore

isabelcosta commented:
@shades-7 this GIF you posted looks amazing 🤩  I also love the feedback you provided, is so specific! To make this awesome I would also suggest using our QA template, just to explain a little of what you tested. You can find the template at point 7 here https://github.com/anitab-org/documentation/blob/master/quality-assurance.md#how-to-test-a-pr

cc @rpattath 🤩

mtreacy002 commented:
@shades-7 , I agree with @isabelcosta that you've done amazing work reviewing the PR. You've raised very good points on your feedback above. As our Github workflow would go from PR submission to PR review then PR testing, I would suggest you post your feedback as the PR review using the `Request changes` option (see screenshot below) so that @naveen8801 could give her response to your feedback and make necessary changes if required. This way, your pr review will also count towards OSH as [what we consider as `Top Participants` are contributors who help Review other's PRs, test other's PRs, or help other participants in general](https://anitab-org.zulipchat.com/#narrow/stream/213491-announcements/topic/Open.20Source.20Hack/near/230741560) 😉.

shades-7 reviewed (changes requested):
* All the social icons are directing to the same page but ideally, they should open the required sites.
* Line spacing should be a little less for this paragraph "We are known for connecting top talents with top comapnies. We make great possibilities happen."
* In mobile view, stacking is done but I think stacking need to de done in all the subsections too like explore, visit etc

naveen8801 commented:
hi @shades-7 @mtreacy002 , where to get all links like social links , legal pages and explore sections ? :-)

Rahulm2310 commented:
@mtreacy002 @isabelcosta  for social links, we are going to add the links to AnitaB.org's social media links. right? Also, as we don't have the designs ready for legal pages i.e. `Terms & Conditions`, `Privacy Policy`, we can directly add links to these pages for now:
1. [Privacy Policy](https://anitab.org/privacy-policy/)
2. [Terms of Use](https://anitab.org/terms-of-use/)
3. [Code of Conduct](https://github.com/anitab-org/documentation/blob/master/Contributing/CODE_OF_CONDUCT.md)

naveen8801 committed:
UI fixes and Social Link added

naveen8801 commented:
hi @Rahulm2310 , i have added social links , let me know if there is some correction :-)

Rahulm2310 commented:
@naveen8801 please wait for @mtreacy002 's point of view.

naveen8801 commented:
> @naveen8801 please wait for @mtreacy002 's point of view.

ok @Rahulm2310  👍

mtreacy002 reviewed (changes requested):
@naveen8801, the page looks amazing 💘 . About the policies links, for now I think we can use the links that @Rahulm2310 's suggested. Although, we could also use the privacy modals @Rahulm2310  created on the register page to keep it consistent across the BridgeInTech application. However, once the issues on the documentation repo on creating [PRIVACY_POLICY.md](https://github.com/anitab-org/documentation/issues/40) and [TERMS_OF_USE.md](https://github.com/anitab-org/documentation/issues/39) are solved, we would change the policies links to those of the documentation repo. So, for now, IMO no need to use the privacy modals as per Register page, let's just use the links given on @Rahulm2310  post above. @naveen8801 can you please add these links? Thanks.

naveen8801 commented:
> @naveen8801, the page looks amazing 💘 . About the policies links, for now I think we can use the links that @Rahulm2310 's suggested. Although, we could also use the privacy modals @Rahulm2310 created on the register page to keep it consistent across the BridgeInTech application. However, once the issues on the documentation repo on creating [PRIVACY_POLICY.md](https://github.com/anitab-org/documentation/issues/40) and [TERMS_OF_USE.md](https://github.com/anitab-org/documentation/issues/39) are solved, we would change the policies links to those of the documentation repo. So, for now, IMO no need to use the privacy modals as per Register page, let's just use the links given on @Rahulm2310 post above. @naveen8801 can you please add these links? Thanks.

hi @mtreacy002 , i think i have already added above links (Privacy Policy and Terms of Use) in my last commit , please have a look  :)

mtreacy002 commented:
Ooops, my bad. Sorry @naveen8801 😄.  Yup, you're right. The links to the privacy policy and terms of use are already there. Ok, will continue with the review now, 😉

naveen8801 commented:
> Ooops, my bad. Sorry @naveen8801 😄.  Yup, you're right. The links to the privacy policy and terms of use are already there. Ok, will continue with the review now, 😉 

No problem 👍😃 @mtreacy002

mtreacy002 reviewed (commented):
@naveen8801 , I tried to run the application after pulling it from your pr branch, but although the application compiled successfully, I'm facing 2 issues:
1. I couldn't see the footer as per your gif. I only can see the existing homepage. 
<img width="1196" alt="Screen Shot 2021-04-17 at 9 41 18 pm" src="https://user-images.githubusercontent.com/29667122/115111864-b640b600-9fc5-11eb-892b-4fc809592d8f.png">

<img width="1255" alt="Screen Shot 2021-04-17 at 9 35 28 pm" src="https://user-images.githubusercontent.com/29667122/115111832-9b6e4180-9fc5-11eb-9b20-226c3306c6f8.png">

2. I noticed 6 warnings came up in relation to the newly created footer. 
<img width="1201" alt="Screen Shot 2021-04-17 at 9 35 54 pm" src="https://user-images.githubusercontent.com/29667122/115111880-cf496700-9fc5-11eb-9446-091a5516ea07.png">

Could you please suggest what went wrong here?
Thanks beforehand 😉

naveen8801 committed:
Warnings fixed and Footer added on Homepage

naveen8801 commented:
> @naveen8801 , I tried to run the application after pulling it from your pr branch, but although the application compiled successfully, I'm facing 2 issues:
> 
> 1. I couldn't see the footer as per your gif. I only can see the existing homepage.
> 
> <img alt="Screen Shot 2021-04-17 at 9 41 18 pm" width="1196" src="https://user-images.githubusercontent.com/29667122/115111864-b640b600-9fc5-11eb-892b-4fc809592d8f.png">
> 
> <img alt="Screen Shot 2021-04-17 at 9 35 28 pm" width="1255" src="https://user-images.githubusercontent.com/29667122/115111832-9b6e4180-9fc5-11eb-9b20-226c3306c6f8.png">
> 
> 1. I noticed 6 warnings came up in relation to the newly created footer.
> 
> <img alt="Screen Shot 2021-04-17 at 9 35 54 pm" width="1201" src="https://user-images.githubusercontent.com/29667122/115111880-cf496700-9fc5-11eb-9446-091a5516ea07.png">
> 
> Could you please suggest what went wrong here?
> Thanks beforehand 😉

Hii @mtreacy002 , actually i removed Footer component from home page before creating this PR and also those warnings are due to use of target="_blank" without **rel** attribute and can be easily resolved by adding **rel="noopener noreferrer"** attribute in those anchor tags where target attribute is used , i have made the changes , now you can review 👍

mtreacy002 reviewed (changes requested):
@naveen8801, the warnings have disappeared and I can see the footer now 🎉.  I just have one minor request. Can you please set it so that homepage content is maintained (so not partially cut) by the footer? 
This is also because later on when we added all the sctions of the Homepage as per the Figma file, the Footer needs to be fixed at the very bottom.
![ezgif com-gif-maker (45)](https://user-images.githubusercontent.com/29667122/115125914-c2049a80-a00e-11eb-9650-b2355cf67adb.gif)

Thanks

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+                  alt="twitter_logo"
+                ></img>
+              </a>
+              <a href="https://www.facebook.com/AnitaBorgIndia/" target="_blank" rel="noopener noreferrer">
```
Why this is linked to `AntaBorgIndia` 's facebook page? It should be linked to AnitaB's [global facebook page](https://www.facebook.com/AnitaB.0rg/).

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+                  alt="gihub_logo"
+                ></img>
+              </a>
+              <a href="https://www.linkedin.com/company/anitaborg-india/" target="_blank" rel="noopener noreferrer">
```
It should also be linked to AnitaB's global linkedin page.

naveen8801 replied:
OK @Rahulm2310 👍

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+            </ul>
+          </div>
+          <div className="flexbox-item">
+            <h3 className="heading">Visit</h3>
```
I think this section can be removed as we don't have a visit location.

naveen8801 replied:
Hi @Rahulm2310 @mtreacy002  , should i remove this section ?

mtreacy002 replied:
@naveen8801 , I believe @Rahulm2310 has answer your question so you can remove this 😉. @Vuyanzi and @isabelcosta unless you'd prefer this `Visit` to show AnitaB.Org mailing address as per anitab.org website? I believe it would be best to keep this consistent with other projects under AnitaB.Org Open Source.
<img width="777" alt="Screen Shot 2021-04-24 at 11 16 14 am" src="https://user-images.githubusercontent.com/29667122/115942664-ab6ab180-a4ee-11eb-92ef-b82bdf9d102b.png">

Vuyanzi replied:
@mtreacy002 is it possible to have AnitaB.orgs mailing address there?

mtreacy002 replied:
@Vuyanzi , I believe @isabelcosta has given the answer to this in her comment [below](https://github.com/anitab-org/bridge-in-tech-web/pull/228#issuecomment-826307730) which is to not use it atm. We can revisit this later when the app is being prepared for production 😉

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+          <div className="flexbox-item">
+            <img className="logo" src="#" alt="Logo"></img>
+            <h4 className="text">
+              We are known for connecting top talents with top comapnies. We
```
Correct the spelling of `companies`.

naveen8801 replied:
Done @Rahulm2310 👍

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+      <div className="footer-container">
+        <div className="main-flexbox">
+          <div className="flexbox-item">
+            <img className="logo" src="#" alt="Logo"></img>
```
@naveen8801 `<img/>` tags are self closing. No need for closing tags.

naveen8801 replied:
Sure :)

naveen8801 committed:
Facebook social link changed and UI fix

naveen8801 committed:
Typo fixed & linkdin link changed

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+            <ul className="list">
+              <li className="li">
+                <a href="https://anitab.org/terms-of-use/" className="Links" target="_blank" rel="noopener noreferrer"  >
+                  Terms
```
```suggestion
                  Terms of Use
```

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+              </li>
+              <li className="li">
+                <a href="https://anitab.org/privacy-policy/" className="Links" target="_blank" rel="noopener noreferrer">
+                  Privacy
```
```suggestion
                  Privacy Policy
```

naveen8801 replied:
> > I just have one minor request. Can you please set it so that homepage content is maintained (so not partially cut) by the footer?
> > This is also because later on when we added all the sctions of the Homepage as per the Figma file, the Footer needs to be fixed at the very bottom.
> 
> @naveen8801 Please fix this as well

@Rahulm2310 @mtreacy002 ,  I think its working fine now , i have added footer component after Section element in Home.jsx due to which home page properly covering 80vh of height .

Rahulm2310 commented on src/home/Home.jsx:
```diff
@@ -1,8 +1,10 @@
 export default function Home() {
 
     return (
+        <div>
```
Please add proper indentation to the jsx.

naveen8801 replied:
Ok @Rahulm2310 👍

Rahulm2310 commented on src/home/Footer.jsx:
```diff
@@ -0,0 +1,118 @@
+              <Link style={{ textDecoration: "none" }}>
+                <li className="li">FAQ</li>
+              </Link>
+              <Link style={{ textDecoration: "none" }}>
```
You can add this style to a class instead of repeating this inline style for every element.

naveen8801 replied:
Ya sure @Rahulm2310  👍

Rahulm2310 commented:
>I just have one minor request. Can you please set it so that homepage content is maintained (so not partially cut) by the footer?
This is also because later on when we added all the sctions of the Homepage as per the Figma file, the Footer needs to be fixed at the very bottom.

@naveen8801 Please fix this as well

naveen8801 committed:
Inline styles replaced , Link names changed

naveen8801 commented:
> > I just have one minor request. Can you please set it so that homepage content is maintained (so not partially cut) by the footer?
> > This is also because later on when we added all the sctions of the Homepage as per the Figma file, the Footer needs to be fixed at the very bottom.
> 
> @naveen8801 Please fix this as well

@Rahulm2310  ,  @mtreacy002  , i think its working fine now , i have added Footer component just below the Section element in Home.jxs due to which home page is properly covering 80vh of height .

Rahulm2310 commented on src/home/Home.jsx:
```diff
@@ -1,8 +1,10 @@
 export default function Home() {
 
     return (
+        <div>
         <section id="home">
         <div className="header-text">
         <h1>Welcome to<br/> Bridge-in-Tech</h1>
```
```suggestion
        <div>
            <section id="home">
                <div className="header-text">
                    <h1>Welcome to<br/> Bridge-in-Tech</h1>
```

Rahulm2310 commented on src/home/Home.jsx:
```diff
@@ -15,5 +17,7 @@ export default function Home() {
         </p>
         </div>
         </section>
+        <Footer/>
+        </div>
```
```suggestion
                    </p>
                </div>
            </section>
        <Footer/>
        </div>
```

Rahulm2310 commented:
@naveen8801 Have you updated the GIF in the PR description? I can't see the changes

naveen8801 commented:
> @naveen8801 Have you updated the GIF in the PR description? I can't see the changes
yes @Rahulm2310  , i have updated it now :)

naveen8801 committed:
Indentation added

naveen8801 commented:
Hi @Rahulm2310  @mtreacy002  , i am not able to find why these tests are falling :( , Please help !

mtreacy002 commented:
@naveen8801 , we're facing an error on the initial pr on test.yml creation (PR #170) and still trying to figure it out. You don't need to worry about why the test is failing on this PR for now, as it might not be coming from your PR. Once we troubleshoot the cause for #170 failing tests, we could rerun the tests here to see if the tests are really failing or not 😉. cc @jalajcodes and @Rahulm2310

Rahulm2310 commented:
```
FAIL src/tests/Register.test.js
  ● Console

    console.error node_modules/jsdom/lib/jsdom/virtual-console.js:29

      Error: Uncaught [Error: Invariant failed: You should not use <Link> outside a <Router>]
```
@mtreacy002  @naveen8801 Test cases for Register page are failing.

mtreacy002 commented:
> ```
> FAIL src/tests/Register.test.js
>   ● Console
> 
>     console.error node_modules/jsdom/lib/jsdom/virtual-console.js:29
> 
>       Error: Uncaught [Error: Invariant failed: You should not use <Link> outside a <Router>]
> ```
> 
> 
> @mtreacy002  @naveen8801 Test cases for Register page are failing.

@Rahulm2310 , yeah, this gets me confused since the tests passed on @aditmehta9 's pr branch as I mentioned [here](https://github.com/anitab-org/bridge-in-tech-web/pull/170#issuecomment-825394197) 🤔

mtreacy002 reviewed (commented):
@naveen8801 , please check my feedback below 😉.

mtreacy002 commented:
> ```
> FAIL src/tests/Register.test.js
>   ● Console
> 
>     console.error node_modules/jsdom/lib/jsdom/virtual-console.js:29
> 
>       Error: Uncaught [Error: Invariant failed: You should not use <Link> outside a <Router>]
> ```
> 
> @mtreacy002 @naveen8801 Test cases for Register page are failing.

@Rahulm2310 , I opened issue #239 to address the failing tests. Please give your suggestion on the approach we could use.

mtreacy002 commented:
@naveen8801 and @Rahulm2310, I'm putting the status `on hold` to wait for the response from @Vuyanzi and/or the @anitab-org/admins  for the [above question](https://github.com/anitab-org/bridge-in-tech-web/pull/228#discussion_r619559935). PS: I've tested the pr locally and everything looks fine. Only need to wait whether or not to remove the `Visit` info from the footer and will submit the review/test report afterward.

isabelcosta commented:
regarding this comment https://github.com/anitab-org/bridge-in-tech-web/pull/228#discussion_r619559935, I think for now we remove the address and focus on the app-specific content, and leave these legal / real contact information, for later.

cc @Vuyanzi @mtreacy002

mtreacy002 commented:
> regarding this comment [#228 (comment)](https://github.com/anitab-org/bridge-in-tech-web/pull/228#discussion_r619559935), I think for now we remove the address and focus on the app-specific content, and leave these legal / real contact information, for later.
> 
> cc @Vuyanzi @mtreacy002

Thank you, @isabelcosta for the answer 🥰. @naveen8801 , can you please remove `Visit` from the code then. Please don't forget to resolve the conflict before you push another commit. I'll review it once you've done. 😉

naveen8801 commented:
> > regarding this comment [#228 (comment)](https://github.com/anitab-org/bridge-in-tech-web/pull/228#discussion_r619559935), I think for now we remove the address and focus on the app-specific content, and leave these legal / real contact information, for later.
> > cc @Vuyanzi @mtreacy002
> 
> Thank you, @isabelcosta for the answer 🥰. @naveen8801 , can you please remove `Visit` from the code then. Please don't forget to resolve the conflict before you push another commit. I'll review it once you've done. 😉

sure @mtreacy002  , I'll do that 👍

naveen8801 committed:
Visit Section removed

mtreacy002 reviewed (approved):
LGTM, @naveen8801 ❤️ .
Tested this locally. Hre's the test summary.
1. Code review: Done 
2. Tested all possible outcomes
    Expected behaviour: Footer section is shown on the Homepage and responsive to different screen sizes.
    Actual behaviour: As expected.
![ezgif com-gif-maker (53)](https://user-images.githubusercontent.com/29667122/116187279-939e6200-a768-11eb-95ea-8d7a15b6b860.gif)

3. Environments:
    OS: MacOS Catalina 10.15.7
    Browser: Google Chrome

Thank you again @naveen8801 for your contribution. 🎉 🎉 🎉 

Now we only need @Rahulm2310 or @Vuyanzi 's approval before we can merge this to `develop` branch 😉

Vuyanzi reviewed (approved):
Looks good @naveen8801 🎊 thanks for your contribution!

Rahulm2310 reviewed (approved):
LGTM 🎉 💯. Thanks @naveen8801

mtreacy002 commented:
Thanks @Vuyanzi and @Rahulm2310 for your approvals. Merging it now :wink:
